### PR TITLE
Fix guard

### PR DESCRIPTION
**Pull request by ArthurGiannotta**
This fixes the problem regarding the usage of a ConditionalDimension that cannot be evaluated.

### CODE REVIEW

File: tests/test_dimension.py [hunk 1]
```diff
@@ -785,3 +785,32 @@ def test_symbolic_factor(self):
         assert np.all(usave.data[1] == 3)
         assert np.all(usave.data[2] == 5)
         assert np.all(usave.data[3] == 7)
+
+    def test_implicit_dims(self):
+        """
+        Test ConditionalDimension as an implicit dimension for an equation.
+        """
+
+        # This test makes an operator that should create a vector of increasing
```
Comment (btrb) [inline tests/test_dimension.py]: Operator?

File: tests/test_dimension.py [hunk 2]
```diff
@@ -785,3 +785,32 @@ def test_symbolic_factor(self):
...
+        # integers, but stop incrementing when a certain stop value is reached
```
Comment (btrb) [inline tests/test_dimension.py]: `.` at the end ?

File: tests/test_dimension.py [hunk 3]
```diff
@@ -785,3 +785,32 @@ def test_symbolic_factor(self):
...
+
+        shape = (50,)
+        stop_value = 20
+
+        time = Dimension(name='time')
+        f = TimeFunction(name='f', shape=shape, dimensions=[time])
+
+        # The condition to stop incrementing
```
Comment (btrb) [inline tests/test_dimension.py]: `.`

File: tests/test_dimension.py [hunk 4]
```diff
@@ -785,3 +785,32 @@ def test_symbolic_factor(self):
...
+        cond = ConditionalDimension(name='cond',
+                                    parent=time, condition=f[time] < stop_value)
+
+        eqs = [Eq(f.forward, f), Eq(f.forward, f.forward + 1, implicit_dims=[cond])]
+        op = Operator(eqs)
+        op.apply(time_M=shape[0] - 2)
+
+        # Make the same calculation in python to assert the result
```
Comment (btrb) [inline tests/test_dimension.py]: `.` at the end as well

File: tests/test_dimension.py (hunk 2 above)
Comment (ArthurGiannotta) [reply tests/test_dimension.py]: The other comments in other tests do not have a . at the end

File: tests/test_dimension.py (hunk 1 above)
Comment (ArthurGiannotta) [reply tests/test_dimension.py]: Fixed the typo

File: tests/test_dimension.py (hunk 2 above)
Comment (FabioLuporini) [reply tests/test_dimension.py]: you can ignore this, @ArthurGiannota

we only enforce fullstops in numpydoc docstrings. Here it's irrelevant

File: tests/test_dimension.py (hunk 3 above)
Comment (FabioLuporini) [reply tests/test_dimension.py]: no

File: tests/test_dimension.py (hunk 4 above)
Comment (FabioLuporini) [reply tests/test_dimension.py]: unnecessary

**Comment by FabioLuporini** [comment]
@ArthurGiannotta I **really** like this!

bug -> MFE -> patch -> PR   -- awesome !

Merging

**Comment by FabioLuporini** [comment]
btw, next time you can use "ir: Fix guard" or something more precise like "ir: Fix guard() in clustering" as commit message. For now, it's fine.

**Pull request merged**